### PR TITLE
5️⃣ <OrganizationProfile/> List, add, edit, remove Domains

### PR DESCRIPTION
**Pull request by panteliselef**
## Type of change

- [ ] 🐛 Bug fix
- [x] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [x] `@clerk/clerk-js`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/remix`
- [x] `@clerk/types`
- [ ] `@clerk/themes`
- [x] `@clerk/localizations`
- [ ] `@clerk/clerk-expo`
- [ ] `@clerk/backend`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/fastify`
- [ ] `@clerk/chrome-extension`
- [ ] `gatsby-plugin-clerk`
- [ ] `build/tooling/chore`

## Description

- [x] `npm test` runs as expected.
- [x] `npm run build` runs as expected.

**Tests will be added in a following PR**

🚨This PR adds `<2 KB` in the organizationprofile.js chunk 🚨
_This is not by accident_

# Video
https://github.com/clerkinc/javascript/assets/19269911/8f70dceb-9dbb-4112-a6fb-0fdb47d3b039

### CODE REVIEW

**Comment by panteliselef** [comment]
!snapshot

**Comment by clerk-cookie** [comment]
Hey @panteliselef - the snapshot version command generated the following package versions:
| Package | Version |
| --- | --- |
| @clerk/backend | 0.28.0-snapshot.0b0cc93 |
| @clerk/chrome-extension | 0.3.27-snapshot.0b0cc93 |
| @clerk/clerk-js | 4.56.0-snapshot.0b0cc93 |
| eslint-config-custom | 0.3.0 |
| @clerk/clerk-expo | 0.18.18-snapshot.0b0cc93 |
| @clerk/fastify | 0.6.4-snapshot.0b0cc93 |
| gatsby-plugin-clerk | 4.4.5-snapshot.0b0cc93 |
| @clerk/localizations | 1.24.2-snapshot.0b0cc93 |
| @clerk/nextjs | 4.23.3-snapshot.0b0cc93 |
| @clerk/clerk-react | 4.24.0-snapshot.0b0cc93 |
| @clerk/remix | 2.10.0-snapshot.0b0cc93 |
| @clerk/clerk-sdk-node | 4.12.3-snapshot.0b0cc93 |
| @clerk/shared | 0.22.0-snapshot.0b0cc93 |
| @clerk/themes | 1.7.5 |
| @clerk/types | 3.50.0-snapshot.0b0cc93 |

Tip: use the snippet copy button below to quickly install the required packages.
```sh
# @clerk/backend
npm i @clerk/backend@0.28.0-snapshot.0b0cc93
```
```sh
# @clerk/chrome-extension
npm i @clerk/chrome-extension@0.3.27-snapshot.0b0cc93
```
```sh
# @clerk/clerk-js
npm i @clerk/clerk-js@4.56.0-snapshot.0b0cc93
```
```sh
# eslint-config-custom
npm i eslint-config-custom@0.3.0
```
```sh
# @clerk/clerk-expo
npm i @clerk/clerk-expo@0.18.18-snapshot.0b0cc93
```
```sh
# @clerk/fastify
npm i @clerk/fastify@0.6.4-snapshot.0b0cc93
```
```sh
# gatsby-plugin-clerk
npm i gatsby-plugin-clerk@4.4.5-snapshot.0b0cc93
```
```sh
# @clerk/localizations
npm i @clerk/localizations@1.24.2-snapshot.0b0cc93
```
```sh
# @clerk/nextjs
npm i @clerk/nextjs@4.23.3-snapshot.0b0cc93
```
```sh
# @clerk/clerk-react
npm i @clerk/clerk-react@4.24.0-snapshot.0b0cc93
```
```sh
# @clerk/remix
npm i @clerk/remix@2.10.0-snapshot.0b0cc93
```
```sh
# @clerk/clerk-sdk-node
npm i @clerk/clerk-sdk-node@4.12.3-snapshot.0b0cc93
```
```sh
# @clerk/shared
npm i @clerk/shared@0.22.0-snapshot.0b0cc93
```
```sh
# @clerk/themes
npm i @clerk/themes@1.7.5
```
```sh
# @clerk/types
npm i @clerk/types@3.50.0-snapshot.0b0cc93
```

File: packages/clerk-js/src/ui/components/OrganizationProfile/CalloutWithAction.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import type { LocalizationKey } from '../../customizables';
+import { Col, Flex, Link, Text } from '../../customizables';
+
+type CalloutWithActionProps = {
+  text: LocalizationKey;
+  actionLabel?: LocalizationKey;
+  onClick?: (e: MouseEvent<HTMLAnchorElement>) => Promise<any>;
+};
+export const CalloutWithAction = (props: CalloutWithActionProps) => {
```
Comment (SokratisVidros) [inline packages/clerk-js/src/ui/components/OrganizationProfile/CalloutWithAction.tsx:11]: 🔧 We should consider moving this to commons as the assumption is the same UI pattern will be used in multiple screens in the future.

File: packages/clerk-js/src/ui/components/OrganizationProfile/DomainList.tsx [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+  const { navigate } = useRouter();
+
+  const isAdmin = membership?.role === 'admin';
+
+  const domainList = useMemo(() => {
+    if (!domains?.data) {
+      return [];
+    }
+
+    return domains.data.filter(d => {
```
Comment (SokratisVidros) [inline packages/clerk-js/src/ui/components/OrganizationProfile/DomainList.tsx:48]: ❓ Why do we need to filter the `domainList`? Shouldn't FAPI take care of that?

File: packages/clerk-js/src/ui/components/OrganizationProfile/OrganizationSettings.tsx [hunk 3]
```diff
@@ -57,6 +59,32 @@ const OrganizationProfileSection = () => {
   );
 };
 
+const OrganizationDomainsSection = () => {
```
Comment (SokratisVidros) [inline packages/clerk-js/src/ui/components/OrganizationProfile/OrganizationSettings.tsx:62]: 🔧 Consider moving this to a separate file.

**Review by SokratisVidros — APPROVED** [review]

File: packages/clerk-js/src/ui/components/OrganizationProfile/CalloutWithAction.tsx (hunk 1 above)
Comment (panteliselef) [reply packages/clerk-js/src/ui/components/OrganizationProfile/CalloutWithAction.tsx:11]: You have a point. Will update this

**Comment by clerk-cookie** [comment]
This PR has been automatically locked since there has not been any recent activity after it was closed. Please open a new issue for related bugs.

**Pull request merged**